openmeetings-core/src/main/java/org/apache/openmeetings/core/remote/KRoom.java

```diff
@@ -173,36 +163,36 @@ public void startRecording(StreamProcessor processor, Client c) {
 			Optional<StreamDesc> osd = c.getScreenStream();
 			if (osd.isPresent()) {
 				osd.get().addActivity(Activity.RECORD);
-				processor.getClientManager().update(c);
+				streamProcessor.getClientManager().update(c);
 				rec.setWidth(osd.get().getWidth());
 				rec.setHeight(osd.get().getHeight());
 			}
-			rec = processor.getRecordingDao().update(rec);
+			rec = streamProcessor.getRecordingDao().update(rec);
 			// Receive recordingId
 			recordingId = rec.getId();
-			for (final KStream stream : streams.values()) {
-				stream.startRecord(processor);
-			}
+			streamProcessor.getStreamsByRoom(this.getRoomId()).forEach(
+					stream -> stream.startRecord(streamProcessor)
+			);
 
 			// Send notification to all users that the recording has been started
 			WebSocketHelper.sendRoom(new RoomMessage(roomId, u, RoomMessage.Type.RECORDING_TOGGLED));
 			log.debug("##REC:: recording in room {} is started {} ::", roomId, recordingId);
 		}
 	}
 
-	public void stopRecording(final StreamProcessor processor, Client c) {
+	public void stopRecording(Client c) {
 		if (recordingStarted.compareAndSet(true, false)) {
 			log.debug("##REC:: recording in room {} is stopping {} ::", roomId, recordingId);
-			for (final KStream stream : streams.values()) {
-				stream.stopRecord();
-			}
-			Recording rec = processor.getRecordingDao().get(recordingId);
+			streamProcessor.getStreamsByRoom(this.getRoomId()).forEach(
+					stream -> stream.stopRecord()
```

Comment:
this can be replaced with `KStream::stopRecord`

Reply:
I think it is? its calling stream.stopRecord();

Reply:
You can replace the code lines 187-189 with
```
			processor.getStreamsByRoom(this.getRoomId()).forEach(KStream::stopRecord);
```